lib/rspec/core/configuration.rb

```diff
@@ -67,15 +67,19 @@ def self.define_reader(name)
       end
 
       # @private
-      def self.define_aliases(name, alias_name)
+      def self.define_alias(name, alias_name)
         alias_method alias_name, name
         alias_method "#{alias_name}=", "#{name}="
-        define_predicate_for alias_name
+        define_predicate alias_name
       end
 
       # @private
-      def self.define_predicate_for(*names)
-        names.each { |name| alias_method "#{name}?", name }
+      def self.define_predicate(name)
+        class_eval <<-PREDICATE, __FILE__ , __LINE__ + 1
+          def #{name}?
+            !!#{name}
+          end
+        PREDICATE
```

Comment:
We prefer to avoid string based evals if we can avoid them, this should be possible to define without it...

Reply:
Yes, it is possible, using a `send`.
I'm curious: what is the rationale against string based evals for meta programming?
I like that I find them typically clearer than their equivalent `define_method`, they are faster, they also can not depend on local variables via closures

Reply:
(modification made)